ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py

```diff
@@ -1789,63 +1789,85 @@ def underlay(self):
             and (self.network_services_l1 or self.network_services_l2 or self.network_services_l3)
         )
+
+    @cached_property
+    def _overlay_vtep(self) -> bool:
         # Set overlay.vtep to enable VXLAN edge PE features
-        vtep = (
+        return (
             self.overlay_routing_protocol in ["ebgp", "ibgp", "her", "cvx"]
             and (self.network_services_l2 or self.network_services_l3)
             and self.underlay_router
             and self.uplink_type == "p2p"
             and self.vtep
         )
-        # Set overlay.evpn to enable EVPN on the node
-        evpn = (
-            self.bgp
-            and (self.evpn_role in ["client", "server"] or self.mpls_overlay_role in ["client", "server"])
-            and self.overlay_routing_protocol in ["ebgp", "ibgp"]
-            and "evpn" in self.overlay_address_families
-        )
-        # Set overlay.evpn_vxlan and overlay.evpn_mpls to differentiate between VXLAN and MPLS use cases.
-        evpn_vxlan = evpn and self.evpn_encapsulation == "vxlan"
-        evpn_mpls = evpn and self.evpn_encapsulation == "mpls"
-        # Set ipvpn_gateway to trigger ipvpn interworking configuration.
-        ipvpn_gateway = evpn and get(self._switch_data_combined, "ipvpn_gateway.enabled", default=False)
+
+    @cached_property
+    def _overlay_vpn_ipv4(self) -> bool:
         # Set overlay.vpn_ipv4 and vpn_ipv6 to enable IP-VPN configuration on the node.
-        # TODO - separate _overlay_<flag> methods for vpn_ipv4/6 logic
-        vpn_ipv4 = self.bgp and (
-            (self.overlay_routing_protocol == "ibgp" and "vpn-ipv4" in self.overlay_address_families)
-            or ("vpn-ipv4" in get(self._switch_data_combined, "ipvpn_gateway.address_families", default=["vpn-ipv4"]) and ipvpn_gateway)
+        if self.bgp is not True:
+            return False
+
+        return (self.overlay_routing_protocol == "ibgp" and "vpn-ipv4" in self.overlay_address_families) or (
+            "vpn-ipv4" in get(self._switch_data_combined, "ipvpn_gateway.address_families", default=["vpn-ipv4"]) and self._overlay_ipvpn_gateway
         )
-        vpn_ipv6 = self.bgp and (
-            (self.overlay_routing_protocol == "ibgp" and "vpn-ipv6" in self.overlay_address_families)
-            or ("vpn-ipv6" in get(self._switch_data_combined, "ipvpn_gateway.address_families", default=["vpn-ipv4"]) and ipvpn_gateway)
+
+    @cached_property
+    def _overlay_vpn_ipv6(self) -> bool:
+        # Set overlay.vpn_ipv4 and vpn_ipv6 to enable IP-VPN configuration on the node.
+        if self.bgp is not True:
+            return False
+
+        return (self.overlay_routing_protocol == "ibgp" and "vpn-ipv6" in self.overlay_address_families) or (
+            "vpn-ipv6" in get(self._switch_data_combined, "ipvpn_gateway.address_families", default=["vpn-ipv4"]) and self._overlay_ipvpn_gateway
         )
+
+    @cached_property
+    def overlay(self):
```

Comment:
might need to add return type (dict) here.

Reply:
Good remark. While this is not the point of this PR and most of the `eos_designs_facts.py` file is missing return type your comment has been addressed in 585596f